src/pages/Questionnaire.tsx

```diff
@@ -60,67 +60,167 @@ const SubmitButton = styled(NavigationButton).attrs({ as: "a" })`
 
 // types
 
-enum Type {
+export enum Type {
   RADIO = "radio",
   CHECKBOX = "checkbox",
   RANGE = "range",
-  DROPDOWN = "dropdown",
 }
 
 interface Option {
   optionId: number;
   optionLabel: string;
+  points: number;
 }
 
-interface QuesProps {
+export interface Question {
   id: number;
   title: string;
   type: Type;
   options: Option[];
 }
 
+type Responses = Record<number, number | number[]>;
+
+// Function to calculate the score for each option - used inside calculateTotalScore function
+const calculateOptionScore = (
+  points: number,
+  numberOfOptions: number,
+  maxPointsPerQuestion: number
+) => {
+  return (points / numberOfOptions) * maxPointsPerQuestion;
+};
+
+// Function to calculate the total score when the user submits the questionnaire
+const calculateTotalScore = (questions: Question[], responses: Responses) => {
+  const numberOfQuestions = questions.length;
+  // Calculation to determine the max points allotted per question.
+  const maxPointsPerQuestion = 100 / numberOfQuestions;
+  let totalScore = 0;
+
+  questions.forEach((question) => {
+    const response = responses[question.id];
+    let questionScore = 0;
+
+    if (response !== undefined) {
+      // Check if the response is an array (checkbox input) before calculating the score
+      if (Array.isArray(response)) {
+        response.forEach((ele) => {
+          const option = question.options.find((opt) => opt.optionId === ele);
+          if (option) {
+            questionScore += calculateOptionScore(
+              option.points,
+              question.options.length,
+              maxPointsPerQuestion
+            );
+          }
+        });
+        // Score calculation for range and radio inputs
```

Comment:
now, this is a great example of a comment. As a reviewer, it may not be clear to me that why are some responses in array and others are not.